packages/maps/src/index.ts

```diff
@@ -1,21 +1,42 @@
-import { Credentials } from '@carto/toolkit-auth';
+const DEFAULT_SERVER_URL_TEMPLATE = 'https://{user}.carto.com';
 
 class Maps {
-  private _credentials: Credentials;
+  private _username: string;
+  private _apiKey: string;
+  private _serverURL: string;
 
-  constructor(username: string, apiKey: string) {
-    this._credentials = new Credentials(username, apiKey);
+  /**
+   * Build an instance to interact with Maps API v1 (aka Windshaft)
+   * @param username
+   * @param apiKey
+   * @param serverURL A url pattern like default ('https://{user}.carto.com')
+   */
+  constructor(username: string, apiKey: string, serverURL: string = DEFAULT_SERVER_URL_TEMPLATE) {
+    this._username = username;
+    this._apiKey = apiKey;
+    this._serverURL = serverURL;
   }
 
-  public get credentials(): Credentials {
-    return this._credentials;
+  public get username(): string {
+    return this._username;
   }
 
-  public set credentials(value: Credentials) {
-    this._credentials = value;
+  public get apiKey(): string {
+    return this._apiKey;
   }
 
-  public instantiateMap(options: { sql?: string, dataset?: string }) {
+  public get serverURL(): string {
+    return this._serverURL;
+  }
+
+  public instantiateMap(options: any) {
+    const mapParameters = this.getMapParameters(options);
+    const requestOptions = this.getRequestOptions(mapParameters);
+
+    return this.fetchLayerGroup(requestOptions).then(this.getVectorTileURL);
```

Comment:
Is this working OK?

As we're using `this.getVectorTileURL` function's reference and using this within the function, I think it would fail :(

Reply:
🤦‍♂